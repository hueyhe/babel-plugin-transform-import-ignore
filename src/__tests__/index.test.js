import { transform } from '@babel/core';
import plugin from '../';

const exampleCode = `
import defaultExport from 'module-name';
import * as name from 'module-name';
import { export1 } from 'module-name';
import { export1 as alias1 } from 'module-name';
import { export3, export4 } from 'module-name';
import { foo , bar } from 'module-name/path/to/specific/un-exported/file';
import { export5 , export6 as alias2 } from 'module-name';
import defaultExport2, { export7 } from 'module-name';
import defaultExport3, * as defaultName from 'module-name';
import 'module-name';

import('./path/to/style.css').then(() => {});

require('./path/to/style.css');

const style = require('./path/to/style.css');

async function dynamicImport() {
  await import('./path/to/style.css');
  const style = await import('./path/to/style.css');
}
`;

it('works', () => {
  const { code } = transform(exampleCode, {
    plugins: [
      [
        plugin,
        {
          patterns: [/\.s?css$/, 'module-name'],
        },
      ]
    ],
  });
  expect(code).toMatchSnapshot();
});
