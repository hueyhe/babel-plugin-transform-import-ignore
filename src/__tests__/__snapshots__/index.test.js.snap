// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"\\"use strict\\";

var name = _interopRequireWildcard();

var defaultName = name;

var _file = require(\\"module-name/path/to/specific/un-exported/file\\");

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== \\"object\\" && typeof obj !== \\"function\\") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const style = require('./path/to/style.css');

async function dynamicImport() {
  const style = await Promise.resolve().then(() => _interopRequireWildcard());
}"
`;
